src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -572,4 +576,34 @@ default boolean canSee(Entity entity) {
         return getValue(Keys.HAS_GRAVITY).get();
     }
 
+    /**
+     * Gets the current display name for this {@link Entity}.
+     *
+     * <p>The display name is determined by the following order:
+     * <ul>
+     *     <li>The display name set through data</li>
```

Comment:
Why would these differ?

Reply:
It's popular on large servers for YouTubers to play disguised as another, or admins to disguise as regular players, or be nicknamed by chat plugins (before mojangs name changes anyway) it's still very popular after those changes as well.